extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -99,12 +100,13 @@ export class AmpAd3PImpl extends AMP.BaseElement {
   }
 
   renderOutsideViewport() {
-    const allowRender = allowRenderOutsideViewport(this.element, this.win);
-    if (allowRender !== true) {
-      return allowRender;
+    if (is3pThrottled(this.win)) {
+      return false;
     }
     // Otherwise the ad is good to go.
-    return super.renderOutsideViewport();
+    const elementCheck = getAmpAdRenderOutsideViewport(this.element);
+    return typeof elementCheck == 'number' ?
+      elementCheck : super.renderOutsideViewport();
```

Comment:
I thought I had commented on this before, but I don't see the comment now, so apologies if this is redundant.

In the case that `getAmpAdRenderOutsideViewport` returns a non-numeric value, you can lose that value here.  Previously, if `allowRender` was `false`, it would be returned.  Now if `elementCheck` is `false`, you'll return `super.renderOutsideViewport()`, which is `3`.